src/Components/RouteTabs.tsx

```diff
@@ -6,26 +6,49 @@ import {
   type BaseTabsProps,
 } from "@artsy/palette"
 import { RouterLink, type RouterLinkProps } from "System/Components/RouterLink"
-import { useIsRouteActive } from "System/Hooks/useRouter"
+import { useIsRouteActive, useRouter } from "System/Hooks/useRouter"
 import type * as React from "react"
+import { useMemo } from "react"
 import { useTracking } from "react-tracking"
 
 export const RouteTab: React.FC<
   React.PropsWithChildren<BaseTabProps & RouterLinkProps>
 > = ({ children, to, ...rest }) => {
+  const { router } = useRouter()
   const tracking = useTracking()
 
   const options = {
     exact: rest.exact !== undefined ? rest.exact : true,
   }
 
+  const location = useMemo(() => {
+    const pathname = to as string
+
+    if (!router || typeof to !== "string") {
+      return { pathname, query: {} }
+    }
+
+    return router.createLocation(pathname)
+  }, [router, to])
+
   return (
     <BaseTab
       as={RouterLink}
       // @ts-ignore
       to={to}
       active={useIsRouteActive(to, options)}
-      onClick={() => {
+      onClick={event => {
```

Comment:
This part, manual router pushing + location construction feels like the most potentially error-prone part here, are there other internal page links/anchors besides tabs? I understand the need here, just pointing out there's sometimes edge cases for the more lower level refactors like this.